eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -0,0 +1,154 @@
+function Get-java-OnboardedDocsMsPackages($DocRepoLocation) {
+    $packageOnboardingFiles = "$DocRepoLocation/package.json"
+  
+    $onboardingSpec = ConvertFrom-Json (Get-Content $packageOnboardingFiles -Raw)
+    $allPackages = @{}
+    foreach ($spec in $onboardingSpec) {
+      $spec.packages | ForEach-Object {$allPackages[$_.packageArtifactId] = $null}
+    }
+    return $allPackages
+  }
+  
+function Get-java-DocsMsTocData($packageMetadata, $docRepoLocation) {
+    # Fallback to get package-level readme name if metadata file info does not exist
+    $packageLevelReadmeName = $packageMetadata.Package.Replace('azure-', '');
+
+    # If there is a metadata json for the package use the DocsMsReadmeName from
+    # the metadata function
+    if ($packageMetadata.PSObject.Members.Name -contains "FileMetadata") {
+        $readmeMetadata = &$GetDocsMsMetadataForPackageFn -PackageInfo $packageMetadata.FileMetadata
+        $packageLevelReadmeName = $readmeMetadata.DocsMsReadMeName
+    }
+
+    $packageTocHeader = $packageMetadata.Package
+    if ($packageMetadata.DisplayName) {
+        $packageTocHeader = $packageMetadata.DisplayName
+    }
+
+    $version = ""
+    $isPreview = $false
+    if($packageMetadata.VersionGA -ne "") {
+        $version = $packageMetadata.VersionGA
+    }
+    else {
+        $version = $packageMetadata.VersionPreview
+        $isPreview = $true
+    }
+    $children = Get-Toc-Children -package $packageMetadata.Package -groupId $packageMetadata.GroupId `
+        -version $version -docRepoLocation $docRepoLocation -isPreview $isPreview
+
+    if (!$children) {
+        Write-Host "Did not find the package namespaces for $($packageMetadata.GroupId):$($packageMetadata.Package):$version"
+    }
+    $output = [PSCustomObject]@{
+        PackageLevelReadmeHref = "~/docs-ref-services/{moniker}/$packageLevelReadmeName-readme.md"
+        PackageTocHeader       = $packageTocHeader
+        TocChildren            = $children
+    }
+    return $output
+}
+
+function Get-java-DocsMsTocChildrenForManagementPackages($packageMetadata, $docRepoLocation) {
+    $children = @()
+    foreach ($package in $packageMetadata) {
+        $version = $package.VersionGA
+        $isPreview = $false
+        if($package.VersionPreview) {
+            $version = $package.VersionPreview
+            $isPreview = $true
+        }
+        $packageToc = Get-Toc-Children -package $package.Package -groupId $package.GroupId -version $version `
+            -docRepoLocation $docRepoLocation -isPreview $isPreview
+        $children += $packageToc
+    }
+    # Flatten the children if multiple packages.
+    return @($children)
```

Comment:
Is the effect of this flattening of a single level of nesting?

Reply:
The problem caused by 
```
New-Item "${Somepath}" -ItemType Directory
```